sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/EliminateSQLFunctionNode.scala

```diff
@@ -34,8 +36,8 @@ object EliminateSQLFunctionNode extends Rule[LogicalPlan] {
     plan.transformWithSubqueries {
       case SQLFunctionNode(_, child) => child
       case f: SQLTableFunction =>
-        throw SparkException.internalError(
-          s"SQL table function plan should be rewritten during analysis: $f")
+        throw QueryCompilationErrors.tableValuedArgumentsNotYetImplementedForSqlFunctions(
```

Comment:
Could you explain, please, why do we catch this in the optimizer, but not in the analyzer?

Reply:
Same question. How do we leak `SQLTableFunction` to the optimizer if we fail earlier in the analyzer/parser?

Reply:
Sorry for the late reply.

After taking a look at the work about `Support SQL user-defined functions` led by Allison, I tried to explain:

For the rule of SQL table functions, PR #49471 is in progress. I think we can also adding the `ResolveSQLTableFunctions` rule in `Analyzer` similarly, and throwing an exception if it is judged that `plan.containsPattern(SQL_TABLE_FUNCTION)` is true.
The current content of the PR is based on quick fixes of existing code to optimize exception messages for the not fully implemented `SQLTableFunction`.

If anyone agrees with the former solution, or has a better idea, I'd love to hear your suggestions.